### PR TITLE
add Secret resource

### PR DESCRIPTION
**Pull request by TheoBrigitte**
Towards: https://github.com/giantswarm/roadmap/issues/3522

This PR adds a secret resource into the Alloy helm chart to allow passing secret configuration via helm values.

### CODE REVIEW

**Review by QuantumEnigmaa — APPROVED** [review]
LGTM

File: helm/alloy/templates/secret.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+metadata:
+  name: {{ include "alloy.fullname" . }}
+  labels:
+    {{- include "alloy.labels" . | nindent 4 }}
+    app.kubernetes.io/component: config
+data:
+  {{- if .Values.alloy.alloy.secret.content }}
+  {{- .Values.alloy.alloy.secret.content | nindent 2 }}
+  {{- else }}
+  {{- .Files.Get "config/example.secret" | nindent 2 }}
```
Comment (QuentinBisson) [inline helm/alloy/templates/secret.yaml:13]: I would prefer that you require the `.Values.alloy.alloy.secret.content` to not be empty than provide an incorrect config though

**Comment by QuentinBisson** [comment]
I assume the secrets needs to be mounted via env variables? Could you add an example explaining how here?

File: helm/alloy/templates/secret.yaml (hunk 1 above)
Comment (TheoBrigitte) [reply helm/alloy/templates/secret.yaml]: That's a good point so I did move the secret example into `helm/alloy/examples/mimir-rules`

**Comment by TheoBrigitte** [comment]
> I assume the secrets needs to be mounted via env variables? Could you add an example explaining how here?

Just did that in `helm/alloy/examples/mimir-rules`

> Also, please open the PR upstream as well to make sure this is supported in the long run

Will do

**Review by QuentinBisson — APPROVED** [review]
<3

**Comment by TheoBrigitte** [comment]
Submitted a PR upstream to add support for the Secret resource https://github.com/grafana/alloy/pull/1502

Also updated this PR to align with the proposition made upstream.

**Pull request merged**